src/Cli/func/Actions/LocalActions/CreateFunctionAction.cs

```diff
@@ -309,6 +309,13 @@ public async Task UpdateLanguageAndRuntime()
                     throw new CliException("Selected language doesn't match worker set in local.settings.json." +
                         $"Selected worker is: {_workerRuntime} and selected language is: {workerRuntimeSelected}");
                 }
+
+                if (workerRuntimeSelected == WorkerRuntime.Java)
```

Comment:
This check should happen much sooner and for all init and new. (Maybe in the GlobalSettings init method).

Scenarios to validate:

- func init
- func init --java
- func init --worker-runtime java
- func new
- func new --java
- func new --language java